### ISSUE TITLE
Profile7 fails on travis ubuntu 14.04 with mono 4.8

### ISSUE THREAD

**Issue by enricosada** [fsharp/fsharp#759]
on travis ci (example [build](https://travis-ci.org/enricosada/FSharp.Quotations.Evaluator/builds/253150566) )

works on osx images, but fails on ubuntu

os: Ubuntu 14.04.5 LTS (trusty)
mono: Mono JIT compiler version 4.8.1 (Stable 4.8.1.0/22a39d7 Wed Apr 12 12:00:40 UTC 2017)
xbuild: XBuild Engine Version 14.0

xbuild doesnt find the `fsharp.core` in the expected directory

```
			Target ResolveAssemblyReferences:
/usr/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets:  warning : Reference 'FSharp.Core' not resolved
				For searchpath {CandidateAssemblyFiles}
				Warning: {CandidateAssemblyFiles} not supported currently
				For searchpath {HintPathFromItem}
				Considered /usr/lib/mono/xbuild/../Reference Assemblies/Microsoft/FSharp/.NETCore/3.3.1.0/FSharp.Core.dll, but it does not exist.
				For searchpath {TargetFrameworkDirectory}
				Considered target framework dir /usr/lib/mono/xbuild-frameworks/.NETPortable/v4.5/Profile/Profile7/, assembly named 'FSharp.Core' not found.
				For searchpath {Registry:,v4.5,}
				Considered '/home/travis/build/enricosada/FSharp.Quotations.Evaluator/src/FSharp.Quotations.Evaluator.Portable/{Registry:,v4.5,}/FSharp.Core' as a file, but the file does not exist
				Considered '/home/travis/build/enricosada/FSharp.Quotations.Evaluator/src/FSharp.Quotations.Evaluator.Portable/{Registry:,v4.5,}/FSharp.Core.exe' as a file, but the file does not exist
				Considered '/home/travis/build/enricosada/FSharp.Quotations.Evaluator/src/FSharp.Quotations.Evaluator.Portable/{Registry:,v4.5,}/FSharp.Core.dll' as a file, but the file does not exist
				For searchpath {RawFileName}
				Considered 'FSharp.Core' as a file, but the file does not exist
```

on osx is found in `/Library/Frameworks/Mono.framework/Versions/4.8.1/lib/mono/xbuild/../Reference Assemblies/Microsoft/FSharp/.NETCore/3.3.1.0/FSharp.Core.dll` using same mono version 4.8.1 (mono-4.8.0-branch/22a39d7)

known workaround: use portable profile from `FSharp.Core` package, more info in https://github.com/travis-ci/travis-ci/issues/4650#issuecomment-135199724

/cc @baronfel

**Comment by dsyme** [comment]
This is fixed as part of https://github.com/fsharp/fsharp/pull/755/

Basically we give up building PCLs and take them from the old nuget package if you want them

**Comment by enricosada** [comment]
@dsyme but what is the exact issue? can be fixed without a mono 4.8.2?
atm is badly packaged on ubuntu 14.04 but works on osx, using same mono 4.8.1

**Comment by dsyme** [comment]
@enricosada There may be various issues.  

* The build of fsharp/fsharp is busted, showing the error you mention.  That's what is being fixed

* Anything in Mono packaging of 4.x won't be fixed

* We will continue to ship in-the-mono-package PCL FSharp.Core, but we should really stop this at some point and make sure everyone is using the FSHarp.Core nuget package.  And migrate people away from PCL towards .NET Standard

**Comment by dsyme** [comment]
@enricosada I've fixed the Mono 5.0 build and integrated it

* Will tag and update the Mono bits

* From now on this repo will assume Mono 5.x, and will assume we are building the repo with ``xbuild``.  

* The resulting compiler bits and runtime bits will still work on Mono 4.x

* I will reverse-integrate the changes to Microsoft/visualfsharp

**Comment by enricosada** [comment]
sry @dsyme didnt want to add work for you. i was just curios why ubuntu doesnt work, and osx does instead (not really into packaging 😄 )

> We will continue to ship in-the-mono-package PCL FSharp.Core, but we should really stop this at some point and make sure everyone is using the FSHarp.Core nuget package

maybe removing these from mono 5 will be good so. is a major version, and make sense

> Anything in Mono packaging of 4.x won't be fixed

Closing as wont fix.

Workaround is use `FSharp.Core` 4.1.* and reference the portable libs, more info in https://github.com/travis-ci/travis-ci/issues/4650#issuecomment-135199724
